noir-projects/aztec-nr/aztec/src/event/event_interface.nr

```diff
@@ -71,48 +157,23 @@ where
     );
     context.push_nullifier(event_commitment);
 
-    context.emit_private_log(log_content, log_content.len());
-}
-
-/// Emits an event as an offchain message. Similar to private log emission but uses offchain message mechanism instead.
-///
-/// Unlike private log emission, encryption here is always unconstrained. This design choice stems from the nature of
-/// offchain messages - they lack guaranteed delivery, unlike private logs. Without delivery guarantees, smart
-/// contracts cannot make assumptions about a message being delivered, making constrained encryption unnecessary.
-/// However, message integrity remains protected through a cryptographic commitment stored in the nullifier tree,
-/// preventing tampering even in the absence of guaranteed delivery. See the description of the
-/// `messages::offchain_message::emit_offchain_message` function for more details on when a guaranteed delivery is
-/// valuable. If guaranteed delivery is required, the `emit_event_in_private_log` function should be used instead.
-pub fn emit_event_as_offchain_message<Event>(
-    event: Event,
-    context: &mut PrivateContext,
-    recipient: AztecAddress,
-)
-where
-    Event: EventInterface + Serialize,
-{
-    // Safety: as explained above, this function places no constraints on the content of the message.
-    let (message_ciphertext, randomness) =
-        unsafe { remove_constraints(|| to_encrypted_private_event_message(event, recipient)) };
+    if emit_as_offchain_message {
+        // In the case of delivery via offchain message tagging is not used and the log content is the ciphertext
+        // itself.
+        let log_content = ciphertext;
 
-    // We generate a cryptographic commitment to the event to ensure its authenticity during out-of-band delivery. Note
-    // that the commitment is made from the (constrained) event content, and not the (unconstrained) ciphertext.
-    // The nullifier tree is chosen over the note hash tree for this purpose since it provides a simpler mechanism
-    // - nullifiers require no nonce, and events, being non-spendable, don't need the guarantee that a "spending"
-    // nullifier can be computed.
-    // TODO(#11571): with decryption happening in Noir we can now use the Packable trait instead.
-    let serialized_event_with_randomness = [randomness].concat(event.serialize());
-    let event_commitment = poseidon2_hash_with_separator(
-        serialized_event_with_randomness,
-        GENERATOR_INDEX__EVENT_COMMITMENT,
-    );
-    context.push_nullifier(event_commitment);
+        // Safety: Offchain messages are by definition unconstrained. They are emitted via the `emit_offchain_effect`
+        // oracle which we don't use for anything besides its side effects, therefore this is safe to call.
+        unsafe { emit_offchain_message(log_content, recipient) };
+    } else {
+        // When delivering through the private log stream, tagging is used so we must prefix the ciphertext with a tag.
```

Comment:
log stream?

Reply:
Since we will need to tackle constrained tagging before mainnet think it makes sense to aim for a final API here and just document this. Added there a [comment](https://github.com/AztecProtocol/aztec-packages/pull/16126/commits/60c6a324f63a01adf01e3c225b493096f83d0ddd) documenting my lies.